src/frontend/app/features/applications/application/application-tabs-base/application-tabs-base.component.html

```diff
@@ -13,22 +14,25 @@ <h1>{{ application?.app?.entity?.name }} </h1>
           <button mat-icon-button name="delete" (click)="redirectToDeletePage()" matTooltip="Delete">
             <mat-icon>delete</mat-icon>
           </button>
-          <div *ngIf="appActions$ | async as appActions" class="app-page-header__actions">
-            <button mat-icon-button name="restart" *ngIf="appActions.restart" (click)="restartApplication()" matTooltip="Restart">
+          <div *ngIf="applicationService.applicationState$ | async as appState" class="app-page-header__actions">
+            <button mat-icon-button name="restart" [disabled]="busy.updating || !appState.actions.restart" (click)="restartApplication()" matTooltip="Restart">
             <mat-icon>settings_backup_restore</mat-icon>
           </button>
-            <button mat-icon-button name="stop" *ngIf="appActions.stop" (click)="stopApplication()" matTooltip="Stop">
+          <button mat-icon-button name="stop" [disabled]="busy.updating" *ngIf="appState.actions.stop;else startButton" (click)="stopApplication()" matTooltip="Stop">
             <mat-icon>stop</mat-icon>
           </button>
-            <button mat-icon-button name="start" *ngIf="appActions.start" (click)="startApplication()" matTooltip="Start">
-            <mat-icon>play_arrow</mat-icon>
-          </button>
-            <button mat-icon-button name="cli" routerLink="/applications/{{applicationService.cfGuid}}/{{applicationService.appGuid}}/cli" matTooltip="CLI Info">
-            <mat-icon>keyboard</mat-icon>
+          <ng-template #startButton>
+            <button mat-icon-button name="start" [disabled]="busy.updating || !appState.actions.start && !appState.actions.stop" (click)="startApplication()" matTooltip="Start">
```

Comment:
I think start button should only be disabled if start is not allowed - [disabled]="busy.updating || !appState.actions.start".